sessionctx/variable/session.go

```diff
@@ -492,11 +492,12 @@ type SessionVars struct {
 	AllowWriteRowID bool
 
 	// AllowBatchCop means if we should send batch coprocessor to TiFlash. Default value is 1, means to use batch cop in case of aggregation and join.
-	// If value is set to 2 , which means to force to send batch cop for any query. Value is set to 0 means never use batch cop.
+	// Value set to 2 means to force to send batch cop for any query. Value set to 0 means never use batch cop.
 	AllowBatchCop int
 
-	// AllowMPPExecution will prefer using mpp way to execute a query.
-	AllowMPPExecution bool
+	// AllowMPPExecution means if we should use mpp way to execute query. Default value is 1 (or 'ON'), means to be determined by the optimizer.
+	// Value set to 2 (or 'ENFORCE') which means to use mpp whenever possible. Value set to 2 (or 'OFF') means never use mpp.
+	AllowMPPExecution string
```

Comment:
emmmmmm, "string" is not a good type to indicate the function of allow mpp and lacks restriction of the var. How about using enum type and name the sub-type after "on", "off", "forced"

Reply:
For the client variable, the reason to use `typeEnum` instead of `typeInt` is to keep compatible with old versions, otherwise, users will get an error after upgrade the cluster. The variable frame will restrict the var value and return a string.

For the internal variable, after our conversation, I set the variable private and add two functions to do it.